File: cus-cms-web/src/api/article.ts
import request from './request'
import type { Article, ArticleCreateReq, ArticleUpdateReq, ArticleFilters } from '@/types/article'
import type { PaginatedData } from '@/types/api'

export const articleApi = {
  getList(params: { page?: number; page_size?: number } & ArticleFilters) {
    return request.get<PaginatedData<Article>>('/articles', { params })
  },
  getDetail(id: string) {
    return request.get<Article>(`/articles/${id}`)
  },
  create(data: ArticleCreateReq) {
    return request.post<Article>('/articles', data)
  },
  update(id: string, data: ArticleUpdateReq) {
    return request.put<Article>(`/articles/${id}`, data)
  },
  remove(id: string) {
    return request.delete(`/articles/${id}`)
  },
  updateStatus(id: string, status: number) {
    return request.put(`/articles/${id}/status`, { status })
  },
}

File: backend/internal/router/article.go
package router

import (
	"novablog/internal/controller"

	"github.com/gin-gonic/gin"
)

// RegisterArticleRoutes 注册文章管理路由。
func RegisterArticleRoutes(r *gin.RouterGroup, articleController *controller.ArticleController, authMiddleware gin.HandlerFunc) {
	articles := r.Group("/articles")
	articles.Use(authMiddleware)
	{
		articles.POST("", articleController.Create)
		articles.GET("", articleController.GetList)
		articles.GET("/:id", articleController.GetDetail)
		articles.PUT("/:id", articleController.Update)
		articles.DELETE("/:id", articleController.Delete)
		articles.PUT("/:id/status", articleController.UpdateStatus)
	}
}
